### PR TITLE
3265 org manager user management

### LINKED ISSUE #3265: Allow org managers to invite users to their org.
To allow a self service model similar to cloud.gov, org managers must be able to invite people to the UAA pages user group and give them access to the org they manage.

### PR DESCRIPTION
**Pull request by davemcorwin**
## Organization and User management

Resolves #3265 

### New Features
- All Users can see the organizations to which they belong
- Organization managers can
  - invite new and existing users to the organization
  - edit the role of organization users
  - remove organization users

### Notes
- Frontend tests are knowingly sparse and we cannot fully test React Hook atm.
- We need to build out some of this in the admin as well

### Questions
- should a manager be able to remove themselves from or edit their own role in an organization?

#### A user that doesn't belong to any organizations

Screen Shot 2021-06-29 at 3 12 57 PM

#### A user that belongs to `agency1` organization and is a manager of `agency2` organization
Screen Shot 2021-06-29 at 3 08 59 PM

#### The "edit" organization screen
Screen Shot 2021-06-29 at 3 09 16 PM

#### Adding a new member
Screen Shot 2021-06-29 at 3 10 31 PM

#### Inviting a new user to the platform
Screen Shot 2021-06-29 at 3 10 48 PM

### CODE REVIEW

**Comment by apburnes** [comment]
Woo!! Lots to dig into but for...
>should a manager be able to remove themselves from or edit their own role in an organization?

If we keep parity with [CF Org Managers permissions](https://docs.cloudfoundry.org/concepts/roles.html#oss-roles-permissions), that would allow them to assign/remove user roles and more customer autonomy.

**Comment by davemcorwin** [comment]
Let's remove the ability for menagers to update the org name

**Review by apburnes — COMMENTED** [review]
I wasn't able to go through the code extensively, but everything looks good at a glance.  Ran through a few different workflows via the UI and most everything was functioning. I added a couple comments where I ran into trouble.  

Some additional UI suggestions:
- The `edit` button on the right side of the org list item through me off. Maybe just aligned left under the org name?
- For invited users in the org user list, maybe an indicator if they have accepted and logged in yet.
- The ability to remove yourself or change your role from `manager` to `user` seems a bit weird when using the UI. Not sure if that could turn into a foot gun.

Awesome work though!!! Huge lift and you got us to the finish line. 👏 👏 👏

File: frontend/components/organization/Edit.jsx [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    message,
+    title: 'Success',
+    position: 'tr',
+    autoDismiss: 3,
+  });
+}
+
+function showInviteAlert(email, link) {
+  // eslint-disable-next-line no-alert
+  return window.alert(`${email} does not have cloud.gov account yet, please send them the following link to get started:\n\n${link}`);
```
[the remark this answers is not in the document]
Comment (davemcorwin) [reply frontend/components/organization/Edit.jsx:24]: agreed on the 🤞

File: frontend/components/organization/Edit.jsx [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        <h3>Members</h3>
+        <ExpandableArea title="Add user" bordered>
+          <AddUserForm
+            className="well"
+            roleOptions={roleOptions}
+            onSubmit={
+              data => actions.inviteToOrganization(org.id, data)
+            }
+            onSubmitSuccess={
+              ({ member, invite: { email, link } }, reduxDispatch) => {
```
[the remark this answers is not in the document]
Comment (davemcorwin) [reply frontend/components/organization/Edit.jsx:125]: this should be fixed!

**Comment by davemcorwin** [comment]
thanks @apburnes !!!

Updates:
- the org members table shows the current user first
- the current user is no longer able to edit or remove their org role
- org members are sorted by email
- the error you saw when adding a currently existing is fixed

> The edit button on the right side of the org list item through me off. Maybe just aligned left under the org name?
It does look a little weird because it's a bit sparse, but it is the same UI as in the sites list so I'm hesitant to change it in the name of consistency.

> For invited users in the org user list, maybe an indicator if they have accepted and logged in yet.
The best we could do here is indicate that they have logged in by checking for the existence of a token, I'm leaning to pushing this to a separate issue and triaging.

> The ability to remove yourself or change your role from manager to user seems a bit weird when using the UI. Not sure if that could turn into a foot gun.
Removed.

Updated screenshot:
Screen Shot 2021-07-01 at 12 21 26 PM

**Review by apburnes — APPROVED** [review]
🥇

**Pull request merged**